runtime/src/accounts_index.rs

```diff
@@ -1548,21 +1542,24 @@ impl<T: IndexValue> AccountsIndex<T> {
         binned.into_iter().for_each(|(pubkey_bin, items)| {
             let w_account_maps = self.account_maps[pubkey_bin].write().unwrap();
             let mut insert_time = Measure::start("insert_into_primary_index");
-            items.into_iter().for_each(|(pubkey, new_item)| {
-                if let InsertNewEntryResults::ExistedNewEntryNonZeroLamports =
-                    w_account_maps.insert_new_entry_if_missing_with_lock(pubkey, new_item)
-                {
-                    // zero lamports were already added to dirty_pubkeys above
-                    dirty_pubkeys.push(pubkey);
-                }
-            });
+            w_account_maps.startup_insert_only(slot, items.into_iter());
             insert_time.stop();
             insertion_time.fetch_add(insert_time.as_us(), Ordering::Relaxed);
         });
 
         (dirty_pubkeys, insertion_time.load(Ordering::Relaxed))
     }
 
+    pub fn get_startup_duplicates(&self) -> Vec<Vec<(Slot, Pubkey)>> {
+        (0..self.bins())
+            .into_iter()
+            .map(|pubkey_bin| {
+                let r_account_maps = self.account_maps[pubkey_bin].read().unwrap();
+                r_account_maps.get_startup_duplicates()
+            })
+            .collect()
+    }
```

Comment:
Feels like the implementation details get leaked out here by returning a `Vec<Vec<<_>>`. Can you do the `.flatten()` here before calling `.collect()`? Or better yet, `.flat_map()` instead of `.map()`?

Reply:
documented. Doesn't seem useful to make extra copies. Caller just iterates through.